### DISCUSSION TITLE
Error when creating a new case

### DISCUSSION THREAD

**Discussion by YidongTong** [NGEET/fates-tutorial#62] [Testing] [answered]
Hi developers,

I encountered an error when I followed the tutorial Setup section "Create Scripts". Using the command ./create_elm-fates-singlepoint-1pft_bci_invinit.sh within the container terminal, it started to run and build.
The error came out:
....
_-------------------------------------------------------------------------
 - Prestage required restarts into /output/bci/fates-tutorial-bci-inventory_init.2024-09-17/run
 - Case input data directory (DIN_LOC_ROOT) is /inputdata 
 - Checking for required input datasets in DIN_LOC_ROOT
-------------------------------------------------------------------------
2024-09-17 18:47:11 PRE_RUN_CHECK HAS FINISHED
run command is mpirun  -np 1  -npernode 4 /output/bci/fates-tutorial-bci-inventory_init.2024-09-17/bld/e3sm.exe   >> e3sm.log.$LID 2>&1  
2024-09-17 18:47:11 SAVE_PRERUN_PROVENANCE BEGINS HERE
Starting job script case.run
2024-09-17 18:47:11 SAVE_PRERUN_PROVENANCE HAS FINISHED
2024-09-17 18:47:11 MODEL EXECUTION BEGINS HERE
2024-09-17 18:47:13 MODEL EXECUTION HAS FINISHED
Exception from case_run: ERROR: RUN FAIL: Command 'mpirun  -np 1  -npernode 4 /output/bci/fates-tutorial-bci-inventory_init.2024-09-17/bld/e3sm.exe   >> e3sm.log.$LID 2>&1 ' failed
See log file for details: /output/bci/fates-tutorial-bci-inventory_init.2024-09-17/run/e3sm.log.240917-184710
Submitted job case.run with id None

________________________________________________________
I checked the log info and it shows:
...
ERROR: GETFIL: FAILED to get /paramfiles/fates_params_default-1pft.nc
#0 0x7ff3fb64f682 in ???
https://github.com/NGEET/fates-tutorial/issues/1 0xc7b60f in ???
https://github.com/NGEET/fates-tutorial/issues/2 0xc7b795 in ???
https://github.com/NGEET/fates-tutorial/issues/3 0xb06610 in ???
https://github.com/NGEET/fates-tutorial/issues/4 0x53592d in ???
https://github.com/NGEET/fates-tutorial/issues/5 0x535fee in ???
https://github.com/NGEET/fates-tutorial/issues/6 0x9f8261 in ???
https://github.com/NGEET/fates-tutorial/issues/7 0xa02a60 in ???
https://github.com/NGEET/fates-tutorial/issues/8 0x5357a0 in ???
https://github.com/NGEET/fates-tutorial/issues/9 0x4f7216 in ???
https://github.com/NGEET/fates-tutorial/issues/10 0x4db545 in ???
https://github.com/NGEET/fates-tutorial/issues/11 0x42e573 in ???
https://github.com/NGEET/fates-tutorial/issues/12 0x41c408 in ???
https://github.com/NGEET/fates-tutorial/issues/13 0x42badc in ???
https://github.com/NGEET/fates-tutorial/issues/14 0x7ff3fadb8d09 in ???
https://github.com/NGEET/fates-tutorial/issues/15 0x40bc99 in ???
https://github.com/NGEET/fates-tutorial/issues/16 0xffffffffffffffff in ???
MPI_ABORT was invoked on rank 0 in communicator MPI_COMM_WORLD
with errorcode 1001.

NOTE: invoking MPI_ABORT causes Open MPI to kill all MPI processes.
You may or may not see output from other processes, depending on
exactly when Open MPI kills them.

______________________________________________

I checked the paramfiles within the container elm-fates, it only includes a README.md without the *fates_params_default-1pft.nc'*, I am not sure whether that induced the error.
![image](https://github.com/user-attachments/assets/6056d328-9de9-44a3-8f3e-f8c2247baf3a)

I only found that there is one fates_params_default.cdl under the fates/parameter_files in the github fates page, I import it and modified my create script, but the error changed to :
PIO: FATAL ERROR: Aborting... FATAL ERROR: NetCDF: Unknown file format (file = /paramfiles/fates_params_default.cdl) (/E3SM/externals/scorpio/src/clib/pioc_support.c: 5090)

I am stuck here now and not sure which step is wrong. 
I also wonder how much space it needs (for example, run the bic site) as I am not sure my local disk is sufficient for allocation.

Thanks for your attention! 

Best,
Yidong

**Comment by jamedina09** [comment, marked as answer]
The file fates_params_default.cdl is a human-readable version of a netcdf file. FATES uses a parameter file ending in .nc.

After downloading the file " fates_params_default.cdl ", you need to convert it to .nc (do this inside the folder "paramfiles" in docker):

```bash
ncgen -o fates_params_default.nc fates_params_default.cdl
````
The next step is to define 1 pft, following:

```bash
./FatesPFTIndexSwapper.py --fin=fates_params_default.nc --fout=fates_params_default-1pft.nc
  --pft-indices=1
````
The file ./FatesPFTIndexSwapper.py is located in:

```bash
cp ../E3SM/components/elm/src/external_models/fates/tools/FatesPFTIndexSwapper.py .
````

**Reply by YidongTong** [reply]
Thank you so much! That works to create the fates_params_default-1pft.nc.

**Comment by glemieux** [comment]
Thanks for the question @YidongTong and for the answer @jamedina09.  I've made a new issue #63 to help address this for future users.

**Reply by YidongTong** [reply]
Thanks! After adding the fates_params_default-1pft.nc and modifying the create scripts, i tried again. Then I got an error again.
The exception error and the log info as follows:
[bci_fates-tutorial-bci-inventory_init.2024-09-18_run_e3sm.log.240918-000511.txt](https://github.com/user-attachments/files/17036623/bci_fates-tutorial-bci-inventory_init.2024-09-18_run_e3sm.log.240918-000511.txt)
[Exception from case_run ERROR.txt](https://github.com/user-attachments/files/17036624/Exception.from.case_run.ERROR.txt)

Thank you for your assistance!

**Reply by glemieux** [reply]
Thank you for sharing your logs and error messages.  Would you take a look in your case `run` directory and see if a `lnd.log` file exists?  It will have some other numbers appended to the end of the filename.  If there is, please share with us that file.  You can take a look in the `lnd.log` to see if there is more specific information on the cause of the error.

**Reply by YidongTong** [reply]
Sure! The lnd.log shows "Error reading.  Var:fates_landuse_pprodharv10_forest_mean. " It seems that the problem is still about the params?

[lnd.log.txt](https://github.com/user-attachments/files/17044934/lnd.log.txt)

**Comment by YidongTong** [comment]
Hi @glemieux , I took a look at the ln.log and share the .txt file below. 
The lnd.log shows "Error reading. Var:fates_landuse_pprodharv10_forest_mean. " It seems that the problem is still about the params?
[lnd.log.txt](https://github.com/user-attachments/files/17044934/lnd.log.txt)

**Reply by glemieux** [reply]
@YidongTong this error is stating that the model is looking for particular variable name in the parameter file, but doesn't see it.  Can you open the parameter file and search to see if `fates_landuse_pprodharv10_forest_mean` is present? 

Note that the parameter file that should be used with the tutorial can be found in the `elm-fates` container under the  `/E3SM/components/elm/src/external_models/fates/parameter_files` directory.

**Reply by YidongTong** [reply]
@glemieux Thank you so much!! It solved!
